src/cryptoadvance/specter/wallet.py

```diff
@@ -192,8 +192,9 @@ def json(self):
         }
 
     def save_to_file(self):
-        with open(self.fullpath, "w+") as f:
-            f.write(json.dumps(self.json, indent=4))
+        with self.lock:
+            with open(self.fullpath, "w+") as f:
+                f.write(json.dumps(self.json, indent=4))
```

Comment:
Isn't that also needed in the device class fo the `_update_keys` and `set_type` functions?

Reply:
Thanks! Updated!